modules/Core/pages/panel/users_reports.php

```diff
@@ -89,36 +89,52 @@
-                    // Reported user exists
-                    $user_reported = $reported_user->getDisplayname();
-                    $user_profile = URL::build('/panel/user/' . urlencode($report->reported_id . '-' . $reported_user->data()->username));
-                    $user_style = $reported_user->getGroupStyle();
-                    $user_avatar = $reported_user->getAvatar();
+            if (!$report->reported_id) {
+                $integration = Integrations::getInstance()->getIntegration('Minecraft');
+                if ($integration != null) {
+                    $reported_user = new IntegrationUser($integration, $report->reported_uuid, 'identifier');
+                    if ($reported_user->exists()) {
+                        $reported_user = $reported_user->getUser();
+
+                        $reported_user_profile = URL::build('/panel/user/' . urlencode($reported_user->data()->id . '-' . $reported_user->data()->username));
+                        $reported_user_style = $reported_user->getGroupStyle();
+                        $reported_user_avatar = $reported_user->getAvatar();
+                    } else {
+                        $reported_user_profile = '#';
+                        $reported_user_style = '';
+                        $reported_user_avatar = AvatarSource::getAvatarFromUUID(Output::getClean($report->reported_uuid));
+                    }
+                } else {
+                    $reported_user_profile = '#';
+                    $reported_user_style = '';
+                    $reported_user_avatar = AvatarSource::getAvatarFromUUID(Output::getClean($report->reported_uuid));
                 }
-            }
 
-            if ($user_reported === null) {
-                // Reported user doesn't exist, use their username and uuid
-                $user_reported = Output::getClean($report->reported_mcname);
-                $user_profile = URL::build('/panel/user/' . urlencode($report->reported_id . '-' . $report->reported_mcname));
-                $user_style = '';
-                $user_avatar = AvatarSource::getAvatarFromUUID($report->reported_uuid ?? $report->reported_mcname);
+                $reported_user_name = Output::getClean($report->reported_mcname);
+            } else {
+                $reported_user = new User($report->reported_id);
+
+                $reported_user_name = $reported_user->getDisplayname();
+                $reported_user_profile = URL::build('/panel/user/' . urlencode($report->reported_id . '-' . $reported_user->data()->username));
+                $reported_user_style = $reported_user->getGroupStyle();
+                $reported_user_avatar = $reported_user->getAvatar();
             }
 
             $updated_by_user = new User($report->updated_by);
 
+            $server = null;
+            if ($report->server_id) {
+                $server = Output::getClean(DB::getInstance()->get('mc_servers', ['id', $report->server_id])->first()->name);
```

Comment:
What if the server does not exist or no longer exists?

Reply:
Good point, will add a null check in here